tests/conftest.py

```diff
@@ -39,6 +39,11 @@ def get(self):
     yield TmpdirFactory()
 
 
+@pytest.fixture()
+def setup_git(monkeypatch, tmpdir):
+    # we should not use user configuration
+    monkeypatch.setenv('HOME', tmpdir)
```

Comment:
we had the same situation in pre-commit-hooks and I think this is the wrong solution as it has the potential to *hide* problems with different configurations.  We instead opted to pass `--no-gpg-sign` to any `git commit`s under test: https://github.com/pre-commit/pre-commit-hooks/pull/121

Reply:
this fixture also doesn't scale (adding it to *literally every test*) -- perhaps instead an `autouse` fixture would have been more appropriate

Reply:
I see